Release/src/http/client/http_client_asio.cpp

```diff
@@ -325,32 +325,53 @@ class asio_connection_pool
     boost::asio::io_service& m_io_service;
     const int m_timeout_secs;
     const bool m_start_with_ssl;
-    const std::function<void(boost::asio::ssl::context&)>& m_ssl_context_callback;
+    std::function<void(boost::asio::ssl::context&)> m_ssl_context_callback;
```

Comment:
This is to prevent a dangling reference, because the lifetime of `asio_connection_pool` can be longer than the function object `m_ssl_context_callback` references. That function object comes from `http_client_config` when creating a `http_client` object.